threads-csharp/IntroductionToSemaphoreInCsharp/Tests/UnitTests.cs

```diff
@@ -0,0 +1,41 @@
+using IntroductionToSemaphoreInCsharp;
+using System.Text;
+
+namespace Tests;
+
+[TestClass]
+public class UnitTests
+{
+    private StringBuilder ConsoleOutput { get; } = new StringBuilder();
+    const int SleepDelay = 50;
+
+    [TestInitialize]
+    public void Init()
+    {
+        Console.SetOut(new StringWriter(this.ConsoleOutput));
+        this.ConsoleOutput.Clear();
+    }
+
+    [TestMethod]
+    public async Task GivenSomeConcurrentCode_WhenAccessWithSemaphoreAsyncIsInvoked_ThenAllThreadsShouldExecute()
+    {
+        await ExampleWithSemaphore.AccessWithSemaphoreAsync(SleepDelay);
+        AssertAllThreadsExecuted();
+    }
+
+    [TestMethod]
+    public async Task GivenSomeConcurrentCode_WhenAccessWithSemaphoreSlimAsyncIsInvoked_ThenAllThreadsShouldExecute()
+    {
+        await ExampleWithSemaphoreSlim.AccessWithSemaphoreSlimAsync(SleepDelay);
+        AssertAllThreadsExecuted();
+    }
+
+    private void AssertAllThreadsExecuted()
+    {
+        var result = ConsoleOutput.ToString();
+        for (int i = 0;i < 10;i++)
```

Comment:
Number 10 looks arbitrary here. I understand that's the number of threads run in the code, so make this number available as a constant in the examples classes and then reference it here instead of hard coding number 10.

Reply:
Also the for loop is not properly formatted, so run a bit of formatting through the file (and others if necessary).

Reply:
Done, created a new class to include constants and declared a new const variable to hold the number of threads